File: java/PE015.java
public class PE015
{
    public static void main(String[] args)
    {
        System.out.println(c(20+20, 20));
    }

    private static long c(long a, long b)
    {
        double result = 1;
        if(a-b < b) {
            b = a - b;
        }
        for(int i = 1; i <= b; i++) {
            result *= (a-i+1)*1.0/i;
        }
        return (long)result;
    }
}

File: c/PE015.c
#include <stdio.h>

long long Combination(int n, int m)
{
    if(n-m < m)
        m = n-m;
    int i;
    double result = 1;
    for(i = 0; i < m; i++)
        result *= (n-i)*1.0/(i+1);
    return (long long)result;
}

int main()
{
    printf("%lld\n", Combination(20+20, 20));
    return 0;
}
